run.c

```diff
@@ -267,12 +266,13 @@ void find_next_sync_point_and_drift(int with_fof)
 
         set_global_time(nexttime);
 
-        move_particles(All.Ti_nextoutput);
-
+        /* Do domain decomp and rebuild tree before doing
+         * power spectrum or FoF; if the particles have
+         * moved a long way, performance will be degraded.*/
```

Comment:
What I mean is "if the particles have moved far enough that they should be exchanged, tree walk performance will be degraded, sometimes to the point that the timestep will not actually complete". Particle neutrinos are a specific example that move fast enough that this happens all the time.

The symptom you describe is indeed one of the failure modes I have encountered with this problem.